### DISCUSSION TITLE
how to run `kustomize build` against a kustomize.toolkit.fluxcd.io/v1beta1 manifest in flux2

### DISCUSSION THREAD

**Discussion by thomas-b-jackson** [fluxcd/flux2#1236] [Q&A]
With flux1, I could test my builds locally by running `kustomize build` against a folder containing a hierarchy of child folders that each contain a kustomize.config.k8s.io/v1beta1 manifest.

In this way, I could test a change locally before pushing the commit to the repo flux1 would sync from.

With flux2, I can't find an equivalent test process.

**Comment by stefanprodan** [comment]
Please see:
- What's the difference between Flux and Kustomize https://toolkit.fluxcd.io/faq/#are-there-two-kustomization-types
- How to validate changes before committing in CI https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/scripts/validate.sh

**Reply by thomas-b-jackson** [reply]
thanks stefan,
that form of validation is helpful, but my primary test use cases is testing Json6902 patches applied via kustomize.toolkit.fluxcd.io/v1beta1 manifests which reference a GitRepository sourceRef. any suggestions for testing that?

**Comment by timd73** [comment]
@thomas-b-jackson  - I think I am looking for the same thing as you are, just want to be sure.

I have my ```kustomize.toolkit.fluxcd.io/v1beta1``` Kustomization whose ```path``` refers to somewhere in a different 
 ```GitRepository``` source which contains a ```kustomize.config.k8s.io/v1beta1``` Kustomization.  I am fine with testing that one with ```kustomize build .```

What I want to do is move some overlays into the ```kustomize.toolkit.fluxcd.io/v1beta1``` Kustomization because they are cluster-specific, and leave the Kustomization in the ```GitRepository``` generic.  What I don't know how to do is test the changes to the ```kustomize.toolkit.fluxcd.io/v1beta1``` Kustomization, i.e. run the equivalent of ```kustomize build``` on the flux Kustomization, to see and test overlays that I move there.

Is that what you need, more or less?  Any idea how to accomplish it?

**Reply by thomas-b-jackson** [reply]
that sounds like my use case!

i'll circle back tomorrow and explain the workaround path i took
